### PR TITLE
Minor doc edits to consolidate wording and for better readability.

### PR DESCRIPTION
**Pull request by meichstedt**
/kind documentation

**What this PR does / why we need it**:

While reviewing #4 on the train, I made a couple edits which I found valuable to share even though that PR has been merged already.

**Special notes for your reviewer**:
You are all awesome and deserve having a nice day!

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: content/docs/architecture.md [hunk 1]
```diff
@@ -2,41 +2,41 @@
...
+KUDO is built to help Dev/Ops teams manage day 2 operations of services on Kubernetes, including stateful services through the management of operators. Day 2 in this context refers to the need to support more than just installation. KUDO is built to support upgrades of services along with backup, recovery and the needs of observability. Building Kubernetes operators requires deep knowledge of the underlying service in addition to a deep knowledge of Kubernetes APIs. KUDO is built as an abstraction such that developers can build an operator without focusing on keeping up with the fast pace of Kubernetes API changes. To this end KUDO supports the following:
```
Comment (meichstedt) [inline content/docs/architecture.md:5]: I don't know why GH can't highlight the change here – which is `past` -> `pace`.

K8s might have had a fast past but I'm sure `pace` is meant.

File: content/docs/architecture.md [hunk 2]
```diff
@@ -2,41 +2,41 @@
...
+* Create operator repositories with index files
 
 ## Architecture Diagram
 
 ![KUDO Architecture Diagram](/images/kudo-architecture.jpg?10x20)
 
 ## Components
 
-**kudoctl** or `kubectl-kudo` is a [kubectl plugin](https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/) command-line client. The client is used to aid developers in creating KUDO operators and is used ops teams to manage operators in a Kubernetes cluster. The client is capable of:
+**kudoctl** or `kubectl-kudo` is a [kubectl plugin](https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/) command-line client. The client is built to aid developers in creating KUDO operators and is used by ops teams to manage operators in a Kubernetes cluster. The client is capable of:
```
Comment (alenkacz) [inline content/docs/architecture.md]: now that I am re-reading it, should we actually call this KUDO CLI as we do in other parts of the docs? 🤔 I don't think we ever refer to it as kudoctl

**Review by alenkacz — APPROVED** [review]
Could we rename the heading of the page to "Architecture" from "KUDO architecture"? It just looks strange in the left panel https://kudo.dev/docs/

File: content/docs/architecture.md (hunk 2 above)
Comment (meichstedt) [reply content/docs/architecture.md]: Right; I was confused about this at first too – I was wondering whether we just introduced a `kudoctl` command …

Comment (meichstedt) [reply content/docs/architecture.md]: Changed it to `kubectl-kudo` since it refers to the label on the diagram

**Review by kensipe — APPROVED** [review]

**Pull request merged**